docs/DeveloperGuide.md

```diff
@@ -496,12 +589,75 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
   Use case ends.
 
-* 3a. The given student is invalid/cannot be found.
+* 3a. The provided student's name is invalid/cannot be found.
+
+    * 3a1. EduManage shows an error message.
+
+      Use case resumes at step 3.
+  
+* 3b. Tuition teacher attempts to tag a student with a Subject without first or simultaneously tagging them with a Level.
+
+    * 3b1. EduManage shows an error message.
+
+      Use case resumes at step 3.
+  
+* 3c. Subject/s provided does not correspond with the Level of the student.
+
+    * 3c1. EduManage shows an error message.
+
+      Use case resumes at step 3.
+
+* 3d. No Subject was provided.
+
+    * 3d1. EduManage shows an error message.
+
+      Use case resumes at step 3.
+
+**Use case: Find students with a common attribute (e.g. Level, Subject)**
+
+**MSS**
+
+1. Tuition teacher requests to list students.
+2. EduManage shows a list of students.
+3. Tuition teacher requests to view all students with a common attribute by providing the common attribute.
+4. EduManage displays all students with the common attribute.
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The provided common attribute is invalid.
```

Comment:
Could elaborate what invalid means.

The provided attribute type is invalid (e.g., an unsupported attribute).